### PR TITLE
checker: check for mut val in immutable obj

### PR DESCRIPTION
**Pull request by yuyi98**
This PR check for mut val in immutable obj.

- Check for mut val in immutable obj.
- Add tests.

```vlang
fn main() {
	a1 := [1, 2, 3]
	for mut j in a1 {
		j *= 2
	}
	a2 := [1, 2, 3]!
	for mut j in a2 {
		j *= 2
	}
	m := {'aa': 1, 'bb': 2}
	for _, mut j in m {
		j *= 2
	}
	for mut j in [1, 2, 3] {
		j *= 2
	}
	for mut j in [1, 2, 3]! {
		j *= 2
	}
	for _, mut j in {'aa': 1, 'bb': 2} {
		j *= 2
	}
}

vlib/v/checker/tests/for_in_mut_val_type.vv:3:15: error: `a1` is immutable, it cannot be changed
    1 | fn main() {
    2 |     a1 := [1, 2, 3]
    3 |     for mut j in a1 {
      |                  ~~
    4 |         j *= 2
    5 |     }
vlib/v/checker/tests/for_in_mut_val_type.vv:7:15: error: `a2` is immutable, it cannot be changed
    5 |     }
    6 |     a2 := [1, 2, 3]!
    7 |     for mut j in a2 {
      |                  ~~
    8 |         j *= 2
    9 |     }
vlib/v/checker/tests/for_in_mut_val_type.vv:11:18: error: `m` is immutable, it cannot be changed
    9 |     }
   10 |     m := {'aa': 1, 'bb': 2}
   11 |     for _, mut j in m {
      |                     ^
   12 |         j *= 2
   13 |     }
vlib/v/checker/tests/for_in_mut_val_type.vv:14:15: error: array literal is immutable, it cannot be changed
   12 |         j *= 2
   13 |     }
   14 |     for mut j in [1, 2, 3] {
      |                  ~~~~~~~~~
   15 |         j *= 2
   16 |     }
vlib/v/checker/tests/for_in_mut_val_type.vv:17:15: error: array literal is immutable, it cannot be changed
   15 |         j *= 2
   16 |     }
   17 |     for mut j in [1, 2, 3]! {
      |                  ~~~~~~~~~~
   18 |         j *= 2
   19 |     }
vlib/v/checker/tests/for_in_mut_val_type.vv:20:19: error: map literal is immutable, it cannot be changed
   18 |         j *= 2
   19 |     }
   20 |     for _, mut j in {'aa': 1, 'bb': 2} {
      |                      ~~~~
   21 |         j *= 2
   22 |     }
```

### CODE REVIEW

File: cmd/tools/vtest-all.v [hunk 1]
```diff
@@ -4,17 +4,38 @@ import os
...
+		msg := if ocmd.okmsg != '' { ocmd.okmsg } else { ocmd.line }
+		println(term.colorize(term.green, '>          OK: $msg '))
+	}
+	for fcmd in fails {
+		msg := if fcmd.errmsg != '' { fcmd.errmsg } else { fcmd.line }
+		println(term.colorize(term.red, '>      Failed: $msg '))
+	}
+	if fails.len > 0 {
+		exit(1)
+	}
```
Comment (spytheman) [inline cmd/tools/vtest-all.v:38]: The change in this file is not related to that PR. I do not see a reason for it.

File: vlib/v/checker/checker.v [hunk 2]
```diff
@@ -3089,6 +3089,25 @@ fn (mut c Checker) stmt(node ast.Stmt) {
 					}
 					if node.val_is_mut {
 						value_type = value_type.to_ptr()
+						match node.cond {
+							ast.Ident {
+								if node.cond.obj is ast.Var {
+									obj := node.cond.obj as ast.Var
+									if !obj.is_mut {
+										c.error('`$obj.name` is immutable, cannot mutated',
```
Comment (spytheman) [inline vlib/v/checker/checker.v:3097]:
```v
'`$obj.name` is immutable, declare it with `mut`, if you want to change it'
```
or
```v
'`$obj.name` is immutable, it cannot be changed'
```

File: cmd/tools/vtest-all.v (hunk 1 above)
Comment (yuyi98) [reply cmd/tools/vtest-all.v:38]:
```vlang
C:\Users\yuyi9\v>v test-all
cannot compile `C:\Users\yuyi9\v\cmd\tools\vtest-all.v`:
cmd\tools\vtest-all.v:86:17: error: `commands` is immutable, cannot mutated
   84 | fn (commands []Command) summary() {
   85 |     sw := time.new_stopwatch({})
   86 |     for mut cmd in commands {
      |                    ~~~~~~~~
   87 |         cmd.run()
   88 |     }
```
Just to fix this error.

File: vlib/v/checker/checker.v [hunk 3]
```diff
@@ -3089,6 +3089,25 @@ fn (mut c Checker) stmt(node ast.Stmt) {
...
+											node.cond.pos)
+									}
+								}
+							}
+							ast.ArrayInit {
+								c.error('array literal is immutable, cannot mutated',
```
Comment (spytheman) [inline vlib/v/checker/checker.v:3103]: same - `it cannot be changed`

File: cmd/tools/vtest-all.v (hunk 1 above)
Comment (spytheman) [reply cmd/tools/vtest-all.v:38]: oh, ok

Comment (spytheman) [reply cmd/tools/vtest-all.v:38]: I did not see it because of all the rest of the refactorings, sorry.

File: vlib/v/checker/checker.v (hunk 2 above)
Comment (yuyi98) [reply vlib/v/checker/checker.v:3097]: OK
